feature_engine/selection/drop_psi_features.py

```diff
@@ -342,6 +352,15 @@ def fit(self, X: pd.DataFrame, y: pd.Series = None):
                 f"and {test_df.shape[0]} samples in the test set. "
                 "Please adjust the value of the cut_off or split_frac."
             )
+        if self.threshold == 'auto':
```

Comment:
To follow sklearn conventions, we **cannot** re-assign new values to attributes that are defined by the user when initializing the transformer. In other words, the init attributes can't be modified. Whatever the user set, needs to prevail. 

Also, I think there is a small issue if defining the threshold at this point: below we have the `switch` parameter, that changes the order of basis and test, which, if I understand correctly, would also affect the value of the threshold, because it depends on the sizes of these datasets. 

I suggest, redefining the threshold when we need it, a bit below, when selecting the features, maybe here:
https://github.com/feature-engine/feature_engine/blob/85c82aad1dda3d3d9282f95d26d80c79261a74d5/feature_engine/selection/drop_psi_features.py#L378

something like:

```
if self.threshoold == 'auto':
   threshold == your calculation
else:
   threshold = self.threshold
```
And then here:

https://github.com/feature-engine/feature_engine/blob/85c82aad1dda3d3d9282f95d26d80c79261a74d5/feature_engine/selection/drop_psi_features.py#L395

change `self.threshold` by just `threshold`

Reply:
Also, thinking it further... in order to be able to test that this threshold is returning the values that we expect, would it not be better to capture the calculation in a hidden method? 

this way, we could add a test to calculate that this method returns the expected threshold values.

self._calculate_auto_threshold(self):
   return your calculation

and then we could test this method with different values of N, M, and bins.

What do you think? 

I think it would also leave us in a better shape if later we want to allow the user to modify the value of the percentile. We already have a test to ensure that nothing will be broken